test/common_stub.py

```diff
@@ -20,9 +20,11 @@ def __init__(self):
 
   def BulkWrite(self, data, unused_timeout_ms=None):
     expected_data = self.written_data.pop(0)
+    if type(expected_data) != type(data) and isinstance(expected_data, bytes):
+        expected_data = expected_data.decode('ascii')
     if expected_data != data:
-      raise ValueError('Expected %s, got %s (%s)' % (
-          _Dotify(expected_data), binascii.hexlify(data), _Dotify(data)))
+        raise ValueError('Expected %s got %s (%s)' % (
+          _Dotify(expected_data),  binascii.hexlify(data.encode('ascii')), _Dotify(data)))
```

Comment:
if you're going to use ascii as the encoding, you need to provide an errors that's either replace or ignore. The default is strict and will just raise an extra, confusing, error instead